### ISSUE TITLE
[docs-infra] Conditionally load Tailwind CSS

### ISSUE THREAD

**Issue by oliviertassinari** [mui/base-ui#617]
### Steps to reproduce

Link to live example: (required)

Steps:
1. Open https://master--base-ui.netlify.app/components/react-dialog/
2. Check the dev tools. Get a huge pile of Tailwind CSS variables spam in the devtools

https://github.com/user-attachments/assets/8c134c8a-8b2d-4a0e-b88a-d58d2b169fc7

3. Spend time during critical page loading to load a CSS file that is unused: 

<img width="849" alt="SCR-20240915-dtxd" src="https://github.com/user-attachments/assets/e30637e4-ead1-474b-a2da-213dcfe4e955">

Same in https://mui.com/material-ui/react-dialog/.

---

It's the same behavior in https://mui.com/material-ui/react-dialog/. It has been driving me crazy for the last 18+ months 😄

### Current behavior

Load Tailwind CSS even when not needed.

### Expected behavior

Only load Tailwind CSS file when needed on the page.

### Context

It's not that important for Base UI, but since it's docs-infra, I think that it's important for Material UI to get clarify about the style as a developer, and as a maintainer, to feel the DX of what you are building like a developer will. 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: -

**Comment by onehanddev** [comment]
@michaldudak, would you like me to look into this? If not, please let me know if there are any other issues I can take a look at to get up to speed. Thanks :)

**Comment by michaldudak** [comment]
@vladmoroz redesigns the site, and, AFAIK, plans to use Tailwind CSS even outside demos, so I believe we will have to live with this.

**Comment by vladmoroz** [comment]
Yeah we'll use Tailwind because it's going to be loaded on pretty much every page anyway and is the current lingua franca of CSS.

I also find this `*` slop extremely annoying though and considering using Tailwind v4 alpha instead, which doesn't have this problem and still results in effectively the same styles

Current:

https://github.com/user-attachments/assets/b777435c-b77f-4893-bfb4-8cc361702c0f


v4:

https://github.com/user-attachments/assets/4870ce02-dc51-4f41-833f-552c16c55023

**Comment by vladmoroz** [comment]
Tailwind CSS variables spam is resolved via #771 
Otherwise we'll continue using Tailwind across the board
